### ISSUE TITLE
Add 'transparent' option for backgroud color

### ISSUE THREAD

**Issue by ivan-radic** [ivan-radic/udl-documentation#1]
Support for Notepad++ themes is sub-optimal. Adding 'transparent' option when defining colors would be a big improvement.

**Comment by Cyberchipz** [comment]
Wanted to say I've noticed this. Transparent would at least allow to maintain the overall Default background color for all text documents set by Notepad++.  These should be the "Theme" colors (selected by the user).  So, transparent would actually set, use, or allow one to use the Theme colors... setting those to existing or chosen default theme colors would probably work just as well... So, we could define a color selection as default theme, or call it 'transparent' meaning 'Theme color".  Naturally, this all changes as one adds sections.

When defining Styler for sections, if a user (like me) is using a black background with white text, because the UDL doesn't allow pass through for user Notepad++ global text defaults, so the Theme or default Notepad++ global text styles problematic for portability to other users. So everything looks strange for me, at first:  (defaulting text to a White background with Black text, all within a Theme default black page style for non text space), at least until I define a style that suits that.  It would be nice if UDL also used the same defaults as Theme for its own default coloring.  I haven't used 'C'/C++ in a long time, or I'd make suggestions and dive into the programming myself and be happy to help... alas... I have forgotten more than I know at this point.

(Problem is if other people want to use my UDL file, they'd be required to manually define styles) depending on their default theme colors, this could be a pretty big re-write of the UDL styles!

So, I would like to add, it would be nice if all 'default' default base text styles could be set to a global local style for UDL, at least initially  which would use Notepad++ user picked, global: page, background and text coloring style on initialization of a new UDL definition.  This way users get a custom experience which matches their pre-existing preferences.   So, all UDL defaults for the app, and defaults for the Styler should be set by Parent Program, in this case Notepad++ as optional parameters.

(Though, I without this option,  am seeing some of this as a bit of a feature, it tells me where I haven't yet set a local style for any of a particular section of UDL.)  but, it would be nice, at least to have the option.

I use dark themes due to eye sensitivity.
FWIW, I did set the Folder & Default/Default Style: Styler option to White text on Black Background; but, that didn't populate though the 'Default' Styler Sections n UDL.

**Comment by ivan-radic** [comment]
Thanks for your comment. You described exactly what the problem is. I'd love to finally finish UDL3, and then I could focus on this problem. The thing is, it requires changes in general N++ "theme-ing" code, not just UDL part, so it is postponed for now.

**Comment by Cyberchipz** [comment]
I'm presuming you're saying that they don't expose the values you need to populate UDL3 with the colors needed.   If that's the issue, I've gotten to know another plugin developer, I could inquire if he's found a workaround on this; though based on his project. the color scheme, I'm thinking he has not; but, I'd be willing to inquire.

**Comment by ivan-radic** [comment]
Exactly, but that is a limitation of Scintilla component. It was the coloring of whitespace characters that got me to put this aside. In Scintilla, there is no way to define the style for `\r` and `\n`. These characters are hardcoded to the style of their own (invert of background and default style) which in basic theme creates ugly looking black squares. In my opinion, whitespace should be visible but non-intrusive, and it needs a user changeable style of its own. This is just a personal preference, but I believe the more programmers realize that whitespace is part of the code, the better we all will be.

For now, I will be happy with implementing a `transparent` background option. This would allow users to make something like `solarized` themes, where the same set of keyword colors works on both light and dark background.

http://ethanschoonover.com/solarized

![687474703a2f2f657468616e7363686f6f6e6f7665722e636f6d2f736f6c6172697a65642f696d672f736f6c6172697a65642d6475616c6d6f64652e706e67](https://user-images.githubusercontent.com/12585358/34648527-04e4e752-f39b-11e7-8882-8dcfcd28ffdb.png)

So, in this sense, `transparent` would simply mean "when applying UDL coloring take N++ global theme background color". Obviously, N++ would have to notify UDL when the theme is changed, so UDL could adjust to runtime changes. But I can tame N++ code, it is Scintilla code base that I need to learn better.

And finally, I need to find some free time to do it all :-)

**Comment by Cyberchipz** [comment]
Oh man do I hear you on that last line!  Free time, so... I'll 'try' not take any more of it than I have... and appreciate what I get!
Ah Scintilla, I know little to nothing... except they don't give Np++ a higher priority on issues, which they possibly have earned... but, (sigh) what do I know. Oh, that's right, nothing, about Scintilla.  But, I can relate to 'learn mode', mine is stuck 'ON' and in 'high' except when the missus makes me slow down and stop.
Solarized themes, that's a new one on me re: text.  But, I follow the concept, it would be interesting if that could be built into a font type as part of an OS so viewing would gain a whole new perspective. Nothing like using Windows OS in 'night light' mode, only to realize ya can't read anything, or only half of it because it casts everything in in a red glow by removing 'blue' from screen spectrum.  How about it casting the red glow, but text gets back just enough blue to make it legible.  A kind of smart font handling.  That might not be what you're thinking; but, either way, I think the concept sounds 'fascinating'...?   I work both late nights and day time... (might be my age)... and color and luminescence can be problematic for me... too bright at night, and not bright enough during the day, the same with color saturation.

But, as I try to fix one thing while trying to develop, I get pulled this way, then that... and finally I realized how far I've gotten from where I started,  I had forgotten what I was trying to accomplish.  And here I am doing it to you.   Hey I really appreciate you getting back to me; I'm a patient man (I work with computers) dey so dumb, we gotta tell them everything and they're so demanding of perfection... it could drive one crazy if they let it.

In the meantime... at least I can configure my combo language use with UDL so I get something other than one color (ah, I remember those days).  I prefer everything other than variables, comments, and pure numbers to have a color cast to them.  Mouseovers are nice too, as long as they don't become too intrusive.

Thanks again, an you have a great day/night... life.  Kudos on UDL and what it is so far... I'll drink to that, and to you, thanks for your kindness, time and consideration!
